resource/doc/libvmaf.md

```diff
@@ -54,9 +54,58 @@ Using FFmpeg with libvmaf is very powerful, as you can create complex filters to
 
 ```
 ffmpeg -i main.mpg -i ref.mpg -filter_complex \
-"[0:v]scale=1920x1080:flags=bicubic[main];[main][1:v]libvmaf" -f null -
+"[0:v]scale=1920x1080:flags=bicubic,setpts=PTS-STARTPTS[main];[1:v]setpts=PTS-STARTPTS[ref];[main][ref]libvmaf" -an -f null -
 ```
 
-Here `main.mpg` is a downscaled and encoded video and `ref.mpg` is its reference source at 1080p. The command scales the first input video (`0:v`) and forwards it to VMAF (`libvmaf`) with the label `main`, where it is compared against the second input reference video, `1:v`. Bicubic upsampling is used (also see the [techblog](https://medium.com/netflix-techblog/vmaf-the-journey-continues-44b51ee9ed12) for the recommendation on upsampling methods).
+Here `main.mpg` is a downscaled and encoded video and `ref.mpg` is its reference source at 1080p. The command scales the first input video (`0:v`) and forwards it to VMAF (`libvmaf`) with the label `main`, where it is compared against the label `ref` which corresponds to the second input reference video, `1:v`. We enforce the scaling algorithm to bicubic, since the default scaling algorithm is bilinear (also see the [techblog](https://medium.com/netflix-techblog/vmaf-the-journey-continues-44b51ee9ed12) for the recommendation on upsampling methods). Note that we need to specify the `setpts=PTS-STARTPTS` option to ensure that the PTS is correctly
+aligned. We can also the `-an` option to disregard the presence of audio.
 
-See the [FFmpeg's guide to libvmaf](https://ffmpeg.org/ffmpeg-filters.html#libvmaf), the [FFmpeg Filtering Guide](https://trac.ffmpeg.org/wiki/FilteringGuide) for more examples of complex filters, and the [Scaling Guide](https://trac.ffmpeg.org/wiki/Scaling) for information about scaling and using different scaling algorithms.
+See the [FFmpeg's guide to libvmaf](https://ffmpeg.org/ffmpeg-filters.html#libvmaf), the [FFmpeg Filtering Guide](https://trac.ffmpeg.org/wiki/FilteringGuide) for more examples of complex filters, and the [Scaling Guide](https://trac.ffmpeg.org/wiki/Scaling) for information about scaling and using different scaling algorithms. Please refer to the next section for some additional examples on more complex cases.
+
+### Use `libvmaf.a` with FFmpeg - More examples
+
+The above command line works well for most cases, but additional steps might be necessary.
+
+For example, when the frame rates between reference and encoded video do not match, e.g., when the source is 60 fps and the encode is 30 fps:
+
+```
+ffmpeg -r 1 -i main.mpg -r 2 -i ref.mpg -filter_complex \
```

Comment:
This seems like a weird choice for forcing different frame rates.

Would you not recommend using the fps filter to (deterministically) upsample the framerate of the encoded video?

That is:

```
ffmpeg -i main.mpg -i ref.mpg -filter_complex \
"[0:v]scale=1920x1080:flags=bicubic,fps=fps=60[main];[main][1:v]libvmaf" -f null -
```

In all pixel-based full reference models I've used so far, it was more common to upsample framerates to the reference rather than doing the opposite.

Or would you recommend not using VMAF in this way since it cannot explicitly handle framerate differences between source/degraded video?

Reply:
This is a good question. There are two reasons why this example mentions -r 1 -r 2. 

The first reason is that the input to the encoder is typically the SFR (standard frame rate) source and not the HFR (high frame rate) one, which is then encoded to as an SFR encode. In this case, calculating VMAF for the SFR encode would mean to compare it with the source used to encode it (SFR version of source). The underlying assumption here is that the HFR->SFR conversion is done by dropping frames and hence the select() operation. If we know exactly the temporal downsampling operation/filter being used, that operation can be used instead. Also, note that the VMAF score represents the quality of the encode with respect to its SFR source.

The second reason is that I found it actually much safer to always include -r 1 and -r 1 when the source and the encode are both SFR or HFR, because in some cases, e.g., image-based sources where we feed in the source frames, ffmpeg will assume a nominal fps of 25, which does not match the encode's fps. Other combinations of specifying the fps as intended etc. did not match the expected results.

Long story short, I think it is safer to always include -r 1 and -1 and if there is an HFR->SFR conversion for the encode to use the -r 2 for the source.

However, I do realize that this example might be very confusing to users, since it entails a number of choices and very special cases (calculate VMAF at SFR version of source and encode rather than HFR and the image-based sources case), so we might as well remove this example.

Reply:
Thanks for explaining the reasoning. To me the choice of specifing an unrealistic framerate at input stage seemed confusing, particularly in the ffmepg context.

I think it would rather make sense to include a short description of the typical workflow then: comparing videos with the same framerate. And in the case of framerate conversions during the encoding, you have several choices for calculating VMAF depending on where and how the framerate conversion was performed. If you did it all with the `select` (or, preferably, `fps`) filter within ffmpeg, then there should be no issue reproducing the HFR → SFR conversion in one command.

We should perhaps include a description of the image-based calculation which indeed is more challenging due to the default ffmpeg input framerate. Note that here, one should actually use `-framerate` instead of `-r`. For example:

> When comparing a sequence of images with an original video source, note that ffmpeg assumes a default frame rate of 25 Hz. If you have a 60 fps source and would like to compare it with a 60 fps image sequence, use the `-framerate` option to tell the ffmpeg image demuxer the assumed framerate:
>
>     ffmpeg -framerate 60 -i "img%04d.png"  -i ref.mpg -filter_complex "[main][0:v]libvmaf" -f null -

Reply:
Thanks for the suggestions! let me try out the -framerate option and get back to you about it.